Fix spurious warning from icephys tutorial

ajtritt opened the pull request:
- track VectorIndex objects in DynamicTable
- set Units.electrodes.table before hitting ObjectMapper
  - the ObjectMapper was attemping to retrieve this
    before it was set, creating another bug

## Motivation

fix #714 


## Checklist

- [X] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured the PR description clearly describes problem and the solution?
- [X] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [X] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [X] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

ajtritt committed:
Fix spurious warning from icephys tutorial

- track VectorIndex objects in DynamicTable
- set Units.electrodes.table before hitting ObjectMapper
  - the ObjectMapper was attemping to retrieve this
    before it was set, creating another bug

t-b commented on src/pynwb/core.py:
```diff
@@ -80,6 +80,21 @@ def __init__(self, **kwargs):
+            if p.neurodata_type == neurodata_type:
+                return p
+            p = p.parent
+        return p
```
@ajtritt Can we just do `return`  here? This is easier to grasp IMHO. Or leave it out entirely.

ajtritt replied:
Good catch. `return None` seems even easier to grasp. Do you object?

ajtritt committed:
clean up ancestor retrieval function

- return None if nothing is found
- rename and make more general
- add unit test

t-b commented:
@ajtritt Very nice!
